client/client.go

```diff
@@ -463,7 +463,9 @@ func (cli *Client) dialer() func(context.Context) (net.Conn, error) {
 		case "unix":
 			return net.Dial(cli.proto, cli.addr)
 		case "npipe":
-			return sockets.DialPipe(cli.addr, 32*time.Second)
+			ctx, cancel := context.WithTimeout(ctx, 32*time.Second)
+			defer cancel()
+			return dialPipeContext(ctx, cli.addr)
```

Comment:
🫠 looks like `testutil` has its own separate implementation of all of this 
https://github.com/moby/moby/blob/c29de525b8fe0ef82f38306f4b20223c78a72c2d/testutil/request/npipe_windows.go#L10-L12

https://github.com/moby/moby/blob/c29de525b8fe0ef82f38306f4b20223c78a72c2d/testutil/request/request.go#L197-L224